asv_bench/benchmarks/pandas/join_merge.py

```diff
@@ -0,0 +1,182 @@
+        self.fracofday = Series(self.fracofday, daily_dates)
+        index = date_range(date_index.min(), date_index.max(), freq="D")
+        self.temp = Series(1.0, index)[self.fracofday.index]
+
+    def time_join_non_unique_equal(self):
+        self.fracofday * self.temp
+
+
+class Merge:
+
+    params = [True, False]
+    param_names = ["sort"]
+
+    def setup(self, sort):
+        N = 10000
+        indices = tm.makeStringIndex(N).values
+        indices2 = tm.makeStringIndex(N).values
+        key = np.tile(indices[:8000], 10)
+        key2 = np.tile(indices2[:8000], 10)
+        self.left = DataFrame(
+            {"key": key, "key2": key2, "value": np.random.randn(80000)}
+        )
+        self.right = DataFrame(
+            {
+                "key": indices[2000:],
+                "key2": indices2[2000:],
+                "value2": np.random.randn(8000),
+            }
+        )
+
+        self.df = DataFrame(
+            {
+                "key1": np.tile(np.arange(500).repeat(10), 2),
+                "key2": np.tile(np.arange(250).repeat(10), 4),
+                "value": np.random.randn(10000),
+            }
+        )
+        self.df2 = DataFrame({"key1": np.arange(500), "value2": np.random.randn(500)})
+        self.df3 = self.df[:5000]
+
+    def time_merge_2intkey(self, sort):
+        merge(self.left, self.right, sort=sort)
+
+    def time_merge_dataframe_integer_2key(self, sort):
+        merge(self.df, self.df3, sort=sort)
+
+    def time_merge_dataframe_integer_key(self, sort):
+        merge(self.df, self.df2, on="key1", sort=sort)
+
+    def time_merge_dataframe_empty_right(self, sort):
+        merge(self.left, self.right.iloc[:0], sort=sort)
+
+    def time_merge_dataframe_empty_left(self, sort):
+        merge(self.left.iloc[:0], self.right, sort=sort)
+
+    def time_merge_dataframes_cross(self, sort):
+        merge(self.left.loc[:2000], self.right.loc[:2000], how="cross", sort=sort)
+
+
+class MergeCategoricals:
+    def setup(self):
+        self.left_object = DataFrame(
+            {
+                "X": np.random.choice(range(0, 10), size=(10000,)),
+                "Y": np.random.choice(["one", "two", "three"], size=(10000,)),
+            }
+        )
+
+        self.right_object = DataFrame(
+            {
+                "X": np.random.choice(range(0, 10), size=(10000,)),
+                "Z": np.random.choice(["jjj", "kkk", "sss"], size=(10000,)),
+            }
+        )
+
+        self.left_cat = self.left_object.assign(
+            Y=self.left_object["Y"].astype("category")
+        )
+        self.right_cat = self.right_object.assign(
+            Z=self.right_object["Z"].astype("category")
+        )
+
+    def time_merge_object(self):
+        merge(self.left_object, self.right_object, on="X")
+
+    def time_merge_cat(self):
+        merge(self.left_cat, self.right_cat, on="X")
+
+
+from .pandas_vb_common import setup  # noqa: F401 isort:skip
```

Comment:
Does this set up the random seed even if it is not called?

Reply:
i think so.  this is the pattern used in pandas, not sure why.

Reply:
I am not sure this works correctly, can you re-check?

Reply:
A function with this name is automatically called by ASV.

Reply:
Got it, thanks!